File: pingYouGou-manager-web/src/main/resources/static/js/service/itemCatService.js
// 定义 Service
app.service("itemCatService", function ($http, pingYouGouServer) {

    /**
     * 查询分类
     * @param parentId
     * @returns {*}
     */
    this.findItemCat = function (parentId) {
        let url = `${pingYouGouServer}/itemCat/itemCatInfo.do/${parentId}`;
        return $http.get(url);
    }

    this.addItemCat = function (itemCat) {
        let url = `${pingYouGouServer}/itemCat/item.do`;
        return $http.post(url, itemCat);
    }

    this.deleteItemCat = function (itemCatId) {
        let url = `${pingYouGouServer}/itemCat/item.do/${itemCatId}`;
        return $http.delete(url);
    }

    this.findItemCatByItemCatId = function (itemCatId) {
        let url = `${pingYouGouServer}/itemCat/item.do/${itemCatId}`;
        return $http.get(url);
    }

    // 修改数据
    this.modifyItemCat = function (itemCat) {
        let url = `${pingYouGouServer}/itemCat/item.do`;
        return $http.put(url, itemCat);
    }
})

File: pingYouGou-sellerGoods-interface/src/main/java/com/lemonzuo/sellergoods/service/ItemCatService.java
package com.lemonzuo.sellergoods.service;

import com.lemonzuo.model.ItemCat;

import java.util.List;

/**
 * @author LemonZuo
 * @create 2019-09-09 14:08
 */
public interface ItemCatService {
    /**
     * 根据上级ID查询分类
     * @param parentId
     * @return
     */
    List<ItemCat> findItemCatByParentId(Long parentId);

    /**
     * 新增
     * @param itemCat
     * @return
     */
    ItemCat addItemCat(ItemCat itemCat);

    /**
     * 是否存在
     * @param name
     * @return
     */
    Integer checkItemCatExist(String name);

    /**
     * 删除
     * @param itemCatId
     * @return
     */
    Integer deleteItemCatByItemCatId(Long itemCatId);

    /**
     * 根据分类ID查询分类信息
     * @param itemCatId
     * @return
     */
    ItemCat findItemCatByItemCatId(Long itemCatId);

    /**
     * 修改分类信息
     * @param itemCat
     * @return
     */
    Integer modifyItemCat(ItemCat itemCat);
}
